src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -438,30 +424,37 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
         }
 
         public async Task SwitchToInstalledPackagesAsync() {
+            if (await SetTabAsync(SelectedTab.InstalledPackages)) {
+                await ReloadInstalledAndLoadedPackagesAsync();
+                await ReplaceItemsAsync(SelectedTab.InstalledPackages);
+            }
+        }
+
+        private async Task<bool> SetTabAsync(SelectedTab tab) {
             await _coreShell.SwitchToMainThreadAsync();
-            if (_selectedTab == SelectedTab.InstalledPackages) {
-                return;
+            if (_selectedTab != tab) {
+                _selectedTab = tab;
+                IsLoading = true;
+                return true;
             }
+            return false;
+        }
 
-            _selectedTab = SelectedTab.InstalledPackages;
+        private async Task ReloadAvailablePackagesAsync() {
             IsLoading = true;
-
             await ReloadInstalledAndLoadedPackagesAsync();
-            if (_selectedTab == SelectedTab.InstalledPackages) {
-                IsLoading = false;
-                ReplaceItems(_installedPackages);
-            }
+            // Updating state only for performance reasons
+            await EnsureAvailablePackagesLoadedAsync(updateStateOnly: true);
```

Comment:
>EnsureAvailablePackagesLoadedAsync [](start = 18, length = 34)

I still think that you don't need to call `EnsureAvailablePackagesLoadedAsync` here - `ReloadInstalledAndLoadedPackagesAsync` should take care of updating available packages that are installed and loaded or will trigger `EnsureAvailablePackagesLoadedAsync(false)` if available packages aren't available yet. Do you have a specific scenario where this call is required?